internal/datacoord/compaction.go

```diff
@@ -219,6 +242,8 @@ func (c *compactionPlanHandler) completeCompaction(result *datapb.CompactionResu
 	}
 	// TODO: when to clean task list
 
+	nodeID := c.plans[planID].dataNodeID
+	c.releaseQueue(nodeID)
 	return nil
```

Comment:
Maybe the mutex is still needed here?

Reply:
It's called from mutex wrapper, cannot add mutex again